ksqldb-common/src/main/java/io/confluent/ksql/config/ImmutableProperties.java

```diff
@@ -36,7 +36,7 @@
       .add(KsqlConfig.KSQL_READONLY_TOPICS_CONFIG)
       .add(KsqlConfig.KSQL_SOURCE_TABLE_MATERIALIZATION_ENABLED)
       .add(KsqlConfig.KSQL_HEADERS_COLUMNS_ENABLED)
-      .add(KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED)
+      //.add(KsqlConfig.KSQL_NEW_QUERY_PLANNER_ENABLED) -- to-do: protect in release
```

Comment:
Whats the plan protecting the conf here?

Reply:
As long as the new planner is not completed, we want to disable it and don't allow users to change the config to enable it. -- However, for QTT, we need to allow to change the config so we need to lift the immutability protection for now (I guess the problem is, that we will also need to remove the corresponding test in the release branch when we re-enable the protection..)

Not sure if there is a better way? -- Ideally, we would change QTT that it is not bound to this protection, but not sure how complex this would be and/or if it's worth it?

Reply:
> I guess the problem is, that we will also need to remove the corresponding test in the release branch when we re-enable the protection
I am just assuming that we will always have to validate old-plans and make sure we are compatible. So even though the property should not be changed at runtime we need a way to test both.

An idea would be to trigger a QTT run for KSQL_NEW_QUERY_PLANNER_ENABLED=true tests and one run for false ?